### PR TITLE
Query Greenwave in batches to avoid timeouts

### PR DESCRIPTION
**Pull request by hluk**
Asking Greenwave service to make decision about too many builds (in an
update) can result in the request timing out.

Instead, this big request can be split into smaller ones, each with a
limited number of builds, which are made sequentially.

Signed-off-by: Lukas Holecek <hluk@email.cz>

---

Couple of progress screenshots:

image

image

image

### CODE REVIEW

File: bodhi/server/templates/update.html [hunk 1]
```diff
@@ -110,26 +110,27 @@
...
-    var get_greenwave_information = function(data) {
+    var get_greenwave_information = function(data, subjects, after_success) {
         data.verbose = true;
         $.ajax({
             url: greenwave_api_url + '/decision',
             type: 'POST',
             contentType: 'application/json',
             data: JSON.stringify(data),
             success: function(data) {
+                console.debug(data);
```
Comment (pypingou) [inline bodhi/server/templates/update.html]: Do we want to keep this one?

Comment (hluk) [reply bodhi/server/templates/update.html]: Ooops, nope. Removing.

**Review by Zlopez — CHANGES_REQUESTED** [review]
I don't see any tests for the new functionality

**Comment by cverna** [comment]
As a note Bodhi does also query greenwave from the backend (https://github.com/fedora-infra/bodhi/blob/develop/bodhi/server/models.py#L1968) and the backend code is called for every updates so much more likely to timeout than the frontend.

I think that we should implement this change in backend too.

**Comment by hluk** [comment]
> I don't see any tests for the new functionality

@Zlopez Are there any browser tests? Couldn't find any, not sure how to write automated tests for this.

**Comment by Zlopez** [comment]
@hluk Unfortunately not now. As I see you added tests for backends methods, which is everything I wanted from you.

**Review by Zlopez — APPROVED** [review]

**Comment by hluk** [comment]
@cverna, @pypingou Can you review again please?

**Comment by Zlopez** [comment]
@hluk `pip-pydocstyle` is failing on this:
```
[2019-07-17T09:53:59.872Z] bodhi/server/models.py:1992 in public method `greenwave_subject_batch_size`:
[2019-07-17T09:53:59.873Z]         D200: One-line docstring should fit on one line with quotes (found 3)
```

**Comment by hluk** [comment]
> D200: One-line docstring should fit on one line with quotes (found 3)

Oh, OK, thanks - I couldn't find the error in Jenkins job log.

**Comment by Zlopez** [comment]
Unfortunately still failing:
```
16:10:09  bodhi/server/models.py:1992 in public method `greenwave_subject_batch_size`:
16:10:09          D210: No whitespaces allowed surrounding docstring text
```

**Pull request merged**